### ISSUE TITLE
can't format to json with llama3.1-8b model

### ISSUE THREAD

**Issue by JasonRD** [MLSysOps/MLE-agent#173]
#### Software execution information

MLE-agent version: 0.3.1
System OS version: ubuntu 20.04.6

#### Problem description
in my scene, i use ollama with llama3.1-8b model.  Following the example instructions,  it throw json format error in step 2.
```
  File "/llm/workspace/MLE-agent/mle/agents/advisor.py", line 134, in interact
    self.report = self.suggest(requirement)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/llm/workspace/MLE-agent/mle/agents/advisor.py", line 124, in suggest
    suggestions = json.loads(text)
                  ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
```
ollama chat response's content:

```
Here's an analysis of your dataset and a suggestion for the best ML task/model/algorithm to use.

**Dataset Summary**

I've previewed your CSV file located at `/lll/workspace/test-mle/imdb.csv`. The data appears to be a movie review data
set, containing text reviews with corresponding sentiment labels (positive or negative). Here's a brief summary:

* Number of instances: 50,000
* Feature types: Text (reviews)
* Target type: Binary classification (sentiment)

**Suggested ML Task/Model/Algorithm**

Based on your requirement for a baseline deep learning model and the characteristics of your dataset, I suggest the fo
llowing:

* **Task**: Text Classification (Sentiment Analysis)
* **Model**: Simple Convolutional Neural Network (CNN) or Recurrent Neural Network (RNN) with Embedding Layer
* **Algorithm**: Train a CNN/RNN model on top of pre-trained word embeddings like Word2Vec, GloVe, or BERT

**Reasoning**

1. **Text Classification**: Your dataset contains text reviews with sentiment labels, making it a perfect fit for text
 classification.
2. **Simple Models**: As you're looking for a baseline model, I've suggested simple architectures (CNN/RNN) that are e
asy to implement and require less computational resources.
3. **Embedding Layer**: Using pre-trained word embeddings helps capture the meaning of words in context, improving the
 model's performance.
4. **State-of-the-Art Techniques**: Although not the most complex models, they incorporate some state-of-the-art techniques like convolutional layers or recurrent neural networks.

**Evaluation Metrics**

For evaluating the performance of your baseline model, I recommend using:

* **Accuracy**
* **Precision**
* **Recall**
* **F1-Score**

These metrics will give you a comprehensive understanding of your model's ability to correctly classify sentiment labe
ls.

**Framework/Tools**

I suggest using PyTorch as your deep learning framework due to its ease of use, flexibility, and extensive documentati
on. You can also consider TensorFlow if you're more comfortable with it.

Here's a summary of my suggestion:

```json
{
  "task": "Text Classification",
  "model_or_algorithm": "Convolutional Neural Network (CNN) or Recurrent Neural Network (RNN)",                         "frameworks": ["PyTorch"],
  "reference": [
    "https://arxiv.org/abs/1409.3226", // CNN for text classification
    "https://arxiv.org/abs/1508.06526" // RNN for text classification
  ],
  "evaluation_metric": ["Accuracy", "Precision", "Recall", "F1-Score"],
  "training_method": "Stochastic Gradient Descent (SGD)",                                                               "serving_method": "Serving is not required",
  "device": "GPU",
  "data_summary": "The data provided is a movie review dataset, containing text reviews with sentiment labels.",
  "suggestion": "Based on the user requirement, we suggest you to use a simple CNN/RNN model with embedding layer for
your baseline deep learning model."
}
```                                                                                                                   
Please let me know if this meets your expectations!
```

#### Steps to reproduce the problem

1. select ollama, when `mle new project`
2. run `mle start --model llama3.1`
3. follow instruction until step 2.

#### Expected behavior

#### Other information

Things you tried, stack traces, related issues, suggestions on how to fix it...

**Comment by leeeizhang** [comment]
Thank you! It appears that the LLM response isn't in a valid JSON format. We'll work on refining the prompts or chains for these smaller LLMs (e.g., Llama 3.1 8B) to ensure they can generate pure JSON outputs as we prompt.

**Comment by HuaizhengZhang** [comment]
yes. outputting json format is not stable for small LLMs.

we are trying to find some open source solutions to fix this issue
